shader/shared/bump.wgsl

```diff
@@ -1,9 +1,22 @@
 // SPDX-License-Identifier: Apache-2.0 OR MIT OR Unlicense
 
-// TODO: robust memory (failure flags)
+// Bitflags for each stage that can fail allocation.
+let STAGE_BINNING: u32 = 0x1u;
+let STAGE_TILE_ALLOC: u32 = 0x2u;
+let STAGE_PATH_COARSE: u32 = 0x4u;
+let STAGE_COARSE: u32 = 0x8u;
+
+// This must be kept in sync with the struct in src/render.rs
 struct BumpAllocators {
+    // Bitmask of stages that have failed allocation.
+    failed: atomic<u32>,
+    binning_size: u32,
+    ptcl_size: u32,
```

Comment:
I prefer the size fields be in `Config`, I'm fairly confident the caching will be better reading from uniforms than here, as the cache line will be constantly bouncing.

Reply:
Ok, will change. My thought here was that we’ll only need to update one buffer on rerun of course but your concern makes sense to me.

Reply:
I meant coarse, of course :)